tests/read_group/test_read_group_model.py

```diff
@@ -0,0 +1,176 @@
+from datetime import datetime
+import uuid
+
+from sqlalchemy.exc import IntegrityError
+from unittest.mock import patch
+
+from dataservice.extensions import db
+from dataservice.api.study.models import Study
+from dataservice.api.participant.models import Participant
+from dataservice.api.biospecimen.models import Biospecimen
+from dataservice.api.genomic_file.models import GenomicFile
+from dataservice.api.sequencing_center.models import SequencingCenter
+from dataservice.api.read_group.models import ReadGroup
+from dataservice.api.sequencing_experiment.models import SequencingExperiment
+from tests.utils import IndexdTestCase
+from tests.mocks import MockIndexd, MockResp
+
+
+class ModelTest(IndexdTestCase):
+    """
+    Test database model
+    """
+    def create_read_group(self):
+        """
+        create sequencing_center and
+        sequencing experiment save the above entities to db
+        returns sequencing_experiment_id
+        """
+        sc = SequencingCenter(name="Baylor")
+        se_id = "Test_SequencingExperiment_0"
+        seq_experiment_data = self._make_seq_exp(external_id=se_id)
+        se = SequencingExperiment(
+            **seq_experiment_data, sequencing_center_id=sc.kf_id)
+        sc.sequencing_experiments.extend([se])
+        db.session.add(sc)
+        db.session.commit()
+        ids = {'sequencing_experiment_id': se_id}
+        return ids
+
+    def test_create_and_find_read_group(self):
+        """
+        Test creation of read_group 
+        """
+        self._create_entities()
+        gf = GenomicFile.query.first()
+        rg = ReadGroup(genomic_file_id=gf.kf_id)
+        db.session.add(rg)
+        db.session.commit()
+
+        self.assertEqual(GenomicFile.query.count(), 1)
+        self.assertEqual(ReadGroup.query.count(), 1)
+        self.assertEqual(ReadGroup.query.first().genomic_file, gf)
+
+    def test_no_gf(self):
+        """
+        Test creation of read_group without a genomic_file
+        """
+        rg = ReadGroup()
+        self.assertRaises(IntegrityError, db.session.add(rg))
```

Comment:
This test will always pass
Change to:
```
db.session.add(rg)
with self.assertRaises(IntegrityError):
        db.session.commit()
```

Reply:
It won't pass when the non-null constraint isn't on the genomic_file relation?

Reply:
It'll pass regardless of whether you have a valid foreign key genomic_file_id on the read group. If you were to add this before the test: 
```
self._create_entities()
gf = GenomicFile.query.first()
rg = ReadGroup(genomic_file_id=gf.kf_id)
```
this won't throw an IntegrityError since there is a valid foreign key and the test will still pass. I think we want a test that tests an IntegrityError is only thrown when it should be?

Reply:
If it doesn't throw an IntegretyError, it won't pass?